build.sh

```diff
@@ -273,9 +411,15 @@ EOF
     mount -t devtmpfs chdev "${rootfs_dir}"/dev || mount --bind /dev "${rootfs_dir}"/dev
     mount -t devpts chpts "${rootfs_dir}"/dev/pts
 
-    echo "### Addings sources list and updating"
-    . tweaks/aptsources.sh
-    . tweaks/extrapkgs.sh
+    echo "### Copying files from tweaks folder"
+    cp -a tweaks/* "${rootfs_dir}"
+    
+    echo "### Adjusting fstab"
+    [[ "$BOOT_DEVICE" == 'usb' ]] && cp "${rootfs_dir}"/etc/fstab.usb "${rootfs_dir}"/etc/fstab
+    [[ "$BOOT_DEVICE" == 'hdd' ]] && cp "${rootfs_dir}"/etc/fstab.hdd "${rootfs_dir}"/etc/fstab    
+    
+    echo "### Running apt in chroot"
+    sed -i -e "s/_release_/$release/g" "${rootfs_dir}/etc/apt/sources.list"
 
     chroot "${rootfs_dir}" /bin/bash -c "apt-get -y update"
```

Comment:
Since i get 
`### Running apt in chroot`
`chroot: failed to run command ‘/bin/bash’: Exec format error`
i expect that somewhere there is somewhere a bash used that is not compiled for armhf, but i haven't figured out where.

Reply:
Didn't get this yet on Ubuntu Focal VM. Maybe I need to try with WSL2?

Reply:
This is most probably related to qemu bugs, see: https://github.com/armbian/build/issues/2787

There seems to be problems when chrooting / qemu into newer bullseye or hirsute rootfs from older (buster/focal/older) versions.

Reply:
This is likely not related to my problem, because i'm building buster. 

in the meantime i found a fix for my problem: executing `sudo update-binfmts --enable qemu-arm` 
I have not yet looked up what this does, but based on the name i would guess that it sets a hook somewhere, that tells the running bash/invoked systemcall to execute called binaries built for armhf with the static qemu executable. I'm not sure though if this should have been done automatically when qemu-user-static was installed. Maybe this is really specific to WSL.

Reply:
And also i'm not sure, why this happened with the changed branch, because i didn't change the WSL env and it worked before with compiling the old script 🤔

Reply:
I tried WSL2 with Ubuntu 20.04 (without setting the option) and I dont get errors making the rootfs. But I get some warnings on the kernel which I don't get in Ubuntu Hirsute VM (virtual box).

This might lead to a problem loading the kernel later...
```
  MODPOST vmlinux.symvers
WARNING: modpost: vmlinux.o (.noinit): unexpected non-allocatable section.
Did you forget to use "ax"/"aw" in a .S file?
Note that for example <linux/init.h> contains
section definitions for use in .S files.
```

Using `sudo update-binfmts --enable qemu-arm` does _not_ fix above problem. So I guess using WSL2 is probably unsafe...

Reply:
That's right, this doesn't fix the problem for the vmlinux file, but it did when the build script didn't execute the chroot in the wsl. 

The same Warning appears to me on a kali vm, also on an ubuntu 20.10 vm. So this looks like it's not WSL2 specific. It didn't show up on a debian buster vm. Based on that i would guess, that this is specific to a certain gcc version, where i'm sure debian has an older version.

Reply:
> Based on that i would guess, that this is specific to a certain gcc version, where i'm sure debian has an older version.

Might be, as noted above, I am using Ubuntu Hirsute (21.04) to compile (which probably has newer gcc, installed with `apt install gcc-arm-none-eabi`) and get no error at all.  

I suppose adding a note to the readme, saying that Ubuntu 21.04 with gcc (version 15:9-2019-q4-0ubuntu2) works definitely and all other combination might/might not work is best course of action?

https://packages.ubuntu.com/hirsute/gcc-arm-none-eabi